metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.ts

```diff
@@ -521,4 +548,25 @@ export class AlertsListComponent implements OnInit, OnDestroy {
     this.subgroupTotal = subgroupTotal;
     this.cdRef.detectChanges();
   }
+
+  toggleQueryBuilder() {
+    this.setSelectedTimeRange([this.selectedTimeRange]);
+    if (!this.hideQueryBuilder) {
+      this.hideQueryBuilder = true;
+      this.manualQuery.nativeElement.value = this.queryBuilder.query;
+    } else {
+      this.hideQueryBuilder = false;
+      this.queryBuilder.clearSearch();
+      this.search();
+    }
+  }
+
+  queryForTreeView() {
```

Comment:
I added a function to handle query generation for the tree view. If the query builder is enabled, we use the generateSelect function like we were before. If the manual query is enabled, we are instead using the value provided by the user in the input.